src/overlays/actors/ovl_En_Sellnuts/z_en_sellnuts.c

```diff
@@ -22,99 +98,1072 @@ const ActorInit En_Sellnuts_InitVars = {
+            this->unk_358 -= 0.1f;
+            this->unk_35C -= 0.1f;
+            break;
+    }
+}
+
+Actor* func_80ADB040(GlobalContext* globalCtx) {
+    Actor* actor = globalCtx->actorCtx.actorList[ACTORCAT_NPC].first;
+
+    while (actor != NULL) {
+        if ((actor->id == ACTOR_EN_SELLNUTS) && !ENSELLNUTS_GET_1(actor)) {
+            return actor;
+        }
+        actor = actor->next;
+    }
+
+    return NULL;
+}
+
+f32 func_80ADB08C(GlobalContext* globalCtx) {
+    s32 pad;
+    Player* player = PLAYER;
+    Actor* sp18 = func_80ADB040(globalCtx);
+
+    if (sp18 != NULL) {
+        return Math_Vec3f_DistXZ(&player->actor.world.pos, &sp18->home.pos);
+    }
+
+    return 80.0f;
+}
+
+void func_80ADB0D8(EnSellnuts* this, GlobalContext* globalCtx) {
+    s16 currentFrame = this->skelAnime.animCurrentFrame;
+    s16 frameCount = SkelAnime_GetFrameCount(&D_80ADD990[this->unk_34C].animationSeg->common);
+
+    if (currentFrame == frameCount) {
+        switch (this->unk_340) {
+            case 0x611:
+            case 0x62A:
+                this->unk_34C = 0;
+                func_8013BC6C(&this->skelAnime, D_80ADD990, 0);
+                this->actionFunc = func_80ADBFA0;
+                break;
+
+            case 0x618:
+                if (this->unk_34C == 1) {
+                    this->unk_34C = 0;
+                    func_8013BC6C(&this->skelAnime, D_80ADD990, 0);
+                    this->actionFunc = func_80ADB544;
+                } else {
+                    this->unk_34C = 1;
+                    func_8013BC6C(&this->skelAnime, D_80ADD990, 1);
+                    this->actionFunc = func_80ADBFA0;
+                }
+                break;
+
+            case 0xFF:
+            case 0x60E:
+            case 0x60F:
+            case 0x614:
+            case 0x628:
+                this->unk_34C = 0;
+                func_8013BC6C(&this->skelAnime, D_80ADD990, 0);
+                this->actionFunc = func_80ADB924;
+                break;
+
+            case 0x610:
+            case 0x615:
+            case 0x616:
+            case 0x629:
+                this->unk_34C = 1;
+                func_8013BC6C(&this->skelAnime, D_80ADD990, 1);
+                this->actionFunc = func_80ADB924;
+                break;
+
+            case 0x613:
+            case 0x619:
+                this->unk_34C = 7;
+                func_8013BC6C(&this->skelAnime, D_80ADD990, 7);
+                this->actionFunc = func_80ADBD64;
+                break;
+        }
+    }
+}
+
+void func_80ADB254(EnSellnuts* this, GlobalContext* globalCtx) {
+    s16 currentFrame = this->skelAnime.animCurrentFrame;
+    s16 frameCount = SkelAnime_GetFrameCount(&D_80ADD990[this->unk_34C].animationSeg->common);
+
+    Math_ApproachS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 2, 0xE38);
+    if ((((this->actor.yDistToPlayer < 50.0f) && (this->actor.yDistToPlayer > -50.0f)) ? 1 : 0) &&
+        ((this->actor.xzDistToPlayer < 200.0f) ? 1 : 0)) {
```

Comment:
```suggestion
    if (((this->actor.yDistToPlayer < 50.0f) && (this->actor.yDistToPlayer > -50.0f) &&
        (this->actor.xzDistToPlayer < 200.0f)) {
```
Are the `? 1 : 0` really needed to match? I've never seen that before. If they are needed, then I think `? true : false` would be fine